vuepress/docs/docs/middleware.md

```diff
@@ -154,4 +156,37 @@ a federated graph:
       def apply[R1](original: GraphQL[R1]): GraphQL[R1] =
         federate(original)
     }
-```
\ No newline at end of file
+```
+
+## Cost Estimation
+
+The `queryCost` and `maxCost` wrappers as well as their variants can be used to estimate the cost of a query, however they require a bit 
+more set up to work properly.
+
+These wrappers are in the `CostEstimation` object which also comes with a special directive that can be used out of the box to instrument your
+schema for cost analysis.
+
+Given a schema in which different fields have different costs to execute, either because they require additional network or computing resources, or database access, or they 
+have some other dependency that makes them expensive to compute. You can add the `CostDirective` to your resolver like so:
+
+```scala
+case class SpokenLineArgs(offset: Int, limit: Int)
+
+@GQLDirective(CostDirective(2))
+case class Character(
+  name: String,
+  // Compute a realtime list of all the spoken lines for this character
+  @GQLDirective(CostDirective(100, multipliers = List("limit")))
+  spokenLines: SpokenLineArgs => UIO[List[String]]
+)
+
+case class Query(
+  @GQLDirective(CostDirective(5)) characters: UIO[List[Character]],
+)
+```
+
+In the above example we have provided a couple different examples. For instance, we can add the directive to both types and 
```

Comment:
Can you add an example of using the wrapper? (so that they can see they have to use `costDirective`).